File: packages/shared/types/order.ts
export type OrderStatus = 'pending' | 'paid' | 'fulfilled' | 'cancelled'

export interface OrderItem {
  sku: string
  variantKey: string
  productName: string
  quantity: number
  unitPrice: number
}

export interface Order {
  id: string
  stripeSessionId: string
  status: OrderStatus
  totalAmount: number
  currency: string
  items: OrderItem[]
  createdAt: string
}

File: apps/api/app/schemas/order.py
from pydantic import BaseModel
from typing import List, Optional, Any
from decimal import Decimal
from datetime import datetime
import uuid


class OrderItemOut(BaseModel):
    sku: str
    variant_key: Optional[str] = None
    product_name: str
    quantity: int
    unit_price: Decimal


class OrderOut(BaseModel):
    id: uuid.UUID
    stripe_session_id: str
    status: str
    total_amount: Decimal
    currency: str
    customer_email: Optional[str] = None
    shipping_address: Optional[Any] = None
    awb_code: Optional[str] = None
    courier_name: Optional[str] = None
    items: List[OrderItemOut]
    created_at: datetime
